tests/schemas/test_identifier_schema.py

```diff
@@ -14,231 +14,175 @@
-    assert valid_scheme_identifier == data == schema.dump(data)
-
-
-class CustomExtraRequiredSchema(IdentifierSchema):
-    """Custom schema."""
+        schema.load(invalid_other)
 
-    def __init__(self, **kwargs):
-        """Constructor."""
-        super().__init__(**kwargs)
 
-    extra = Str(required=True)
-
-
-def test_custom_inheritance_extra_provided_should_pass():
-    valid_custom = {
-        "extra": "random",
-        "identifier": "0000-0001-6759-6273",
+def test_detected_and_allowed_scheme_valid_value():  # 5
+    schema = IdentifierSchema(allowed_schemes=["doi"])
+    valid_doi = {
+        "identifier": "10.12345/foo.bar"
     }
+    data = schema.load(valid_doi)
 
-    loaded = CustomExtraRequiredSchema().load(valid_custom)
-    assert valid_custom == loaded == CustomExtraRequiredSchema().dump(loaded)
+    valid_doi["scheme"] = "doi"
+    assert data == valid_doi
 
 
-def test_custom_inheritance_extra_not_provided_but_required_should_fail():
-    invalid_custom = {}
+def test_detected_and_not_allowed_scheme_valid_value():  # 6
+    schema = IdentifierSchema(allowed_schemes=[("other", validate_other)])
+    valid_doi = {
+        "identifier": "10.12345/foo.bar"
+    }
 
     with pytest.raises(ValidationError):
-        CustomExtraRequiredSchema().load(invalid_custom)
-
-
-class CustomExtraNotRequiredSchema(IdentifierSchema):
-    """Custom schema."""
-
-    def __init__(self, **kwargs):
-        """Constructor."""
-        super().__init__(**kwargs)
+        schema.load(valid_doi)
 
-    # Not required to see the failure comes from the identifier
-    extra = Str()
 
-
-def test_custom_inheritance_extra_with_identifier_should_pass():
-    valid_custom = {
-        "extra": "random",
-        "identifier": "0000-0001-6759-6273",
-        "scheme": "orcid",
+def test_detected_and_allowed_scheme_respect_detection_orther():  # 8
+    # NOTE: pmid is PubMed ID
+    schema = IdentifierSchema(allowed_schemes=["orcid", "isni"])
+    valid_orcid = {
+        "identifier": "0000-0001-6759-6273"
     }
+    data = schema.load(valid_orcid)
 
-    loaded = CustomExtraNotRequiredSchema().load(valid_custom)
-    assert (
-        valid_custom == loaded == CustomExtraNotRequiredSchema().dump(loaded)
-    )
-
+    valid_orcid["scheme"] = "orcid"
+    assert data == valid_orcid
 
-def test_custom_inheritance_extra_without_identifier_should_fail():
-    invalid_custom = {}
 
-    with pytest.raises(ValidationError):
-        CustomExtraNotRequiredSchema().load(invalid_custom)
-
-    invalid_custom = {
-        "extra": "random",
+def test_detected_and_allowed_scheme_second_detected():  # 8
+    schema = IdentifierSchema(allowed_schemes=["isni"])
+    valid_isni = {
+        "identifier": "0000-0001-6759-6273"
     }
+    data = schema.load(valid_isni)
 
-    with pytest.raises(ValidationError):
-        CustomExtraNotRequiredSchema().load(invalid_custom)
+    valid_isni["scheme"] = "isni"
+    assert data == valid_isni
```

Comment:
If we want to cover last branch of `IdentifierSchema`'s `__init__`, we could do:

```python
def test_invalid_allowed_schemes():
    with pytest.raises(ValidationError):
        IdentifierSchema(allowed_schemes=[1, 2, 3])
```

Reply:
True, thanks! added at first (along with other more related tests)